PickoGKRuntime fails to compile with clang 15.0.7 on Linux

## Description

Not sure if this is platform dependant, compiler dependant or both, but clang-15 under Ubuntu 22.04 fails to compile.

## Steps to reproduce

```sh
cmake -B build/clang -DCMAKE_C_COMPILER=clang-15 -DCMAKE_CXX_COMPILER=clang++-15
cmake --build build/clang -j
```

### Output

```log
PicoGKRuntime/Source/PicoGKVdbField.h:57:11: error: missing 'typename' prior to dependent type name 'TFieldType::Ptr'
    Field(TFieldType::Ptr roGrid)
          ^~~~~~~~~~~~~~~
          typename 
PicoGKRuntime/Source/PicoGKVdbField.h:67:5: error: missing 'typename' prior to dependent type name 'TFieldType::Ptr'
    TFieldType::Ptr roVdbGrid() const {return m_roGrid;}
    ^~~~~~~~~~~~~~~
    typename 
PicoGKRuntime/Source/PicoGKVdbField.h:70:5: error: missing 'typename' prior to dependent type name 'TFieldType::Ptr'
    TFieldType::Ptr             m_roGrid;
    ^~~~~~~~~~~~~~~
    typename 
3 errors generated.
```

## Additional information

### Compiler information

```sh
❯ clang-15 -v
Ubuntu clang version 15.0.7
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/10
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/11
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/12
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/9
Selected GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/12
Candidate multilib: .;@m64
Selected multilib: .;@m64
❯ uname -a
Linux gn 6.5.0-28-generic #29~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Apr  4 14:39:20 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
```

FYI and for a sanity check on my M1 Mac with clang 15.0.0 (clang-1500.3.9.4) PicoGKRuntime compiles fine.

--- a/Source/PicoGKVdbField.h
+++ b/Source/PicoGKVdbField.h
@@ -54,7 +54,7 @@ public:
         m_roGrid = TFieldType::create();
     };
     
-    Field(TFieldType::Ptr roGrid)
+    Field(typename TFieldType::Ptr roGrid)
     {
         m_roGrid = roGrid;
     };
@@ -64,10 +64,10 @@ public:
         m_roGrid = deepCopyTypedGrid<TFieldType>(oSource.m_roGrid);
     };
     
-    TFieldType::Ptr roVdbGrid() const {return m_roGrid;}
+    typename TFieldType::Ptr roVdbGrid() const {return m_roGrid;}
     
 protected:
-    TFieldType::Ptr             m_roGrid;
+    typename TFieldType::Ptr             m_roGrid;
 };
 
 class ScalarField : public Field<FloatGrid>

